messaging/broker.go

```diff
@@ -92,25 +126,67 @@ func (b *Broker) Open(path string, u *url.URL) error {
 	}
 	b.path = path
 
-	// Require a non-blank connection address.
-	if u == nil {
-		return ErrConnectionAddressRequired
+	// Ensure root directory exists.
+	if err := os.MkdirAll(path, 0700); err != nil {
```

Comment:
@rothrock tells me we should actually be creating directories with permission `0755` as this allows influxd to be started as root, but then launch more instances as another user, and still access the directory.